Sources/_SwiftSyntaxTestSupport/IncrementalParseTestUtils.swift

```diff
@@ -78,13 +85,22 @@ public func assertIncrementalParse(
     return
   }
 
+  var lastRangeUpperBound = originalString.startIndex
   for expectedReusedNode in expectedReusedNodes {
-    guard let range = getByteSourceRange(for: expectedReusedNode.source, in: originalString) else {
+    guard let range = byteSourceRange(for: expectedReusedNode.source, in: originalString[lastRangeUpperBound...]) else {
       XCTFail("Fail to find string in original source,", file: expectedReusedNode.file, line: expectedReusedNode.line)
       continue
     }
 
-    guard let reusedNode = reusedNodes.first(where: { $0.byteRangeAfterTrimmingTrivia == range }) else {
+    guard
+      let reusedNode = reusedNodes.first(where: {
+        $0.byteRangeAfterTrimmingTrivia
+          == ByteSourceRange(
+            offset: range.offset + originalString.distance(from: originalString.startIndex, to: lastRangeUpperBound),
+            length: range.length
+          )
+      })
+    else {
```

Comment:
Instead of doing all these index manipulations (which seem fairly brittle to me), I would keep everything as is and add a `after` parameter to `byteSourceRange`

```swift
fileprivate func byteSourceRange(for substring: String, in sourceString: String, after: String.Index) -> ByteSourceRange? {
  if let range = sourceString[after...].range(of: substring) {
    return ByteSourceRange(
      offset: sourceString.utf8.distance(from: sourceString.startIndex, to: range.lowerBound),
      length: sourceString.utf8.distance(from: range.lowerBound, to: range.upperBound)
    )
  }
  return nil
}
```

You can then call `byteSourceRange` above as

```swift
byteSourceRange(for: expectedReusedNode.source, in: originalString, after: lastRangeUpperBound)
```